### PR TITLE
Convert textinput to use str over char *buffer

### PR DESCRIPTION
**Pull request by mrannanj**

### CODE REVIEW

**Comment by mrannanj** [comment]
This is currently broken, need to look into it ..

File: src/game/gui/textinput.c [hunk 1]
```diff
@@ -20,7 +20,7 @@ typedef struct {
     int ticks;
     int dir;
     surface sur;
-    char *buf;
+    str *text;
```
Comment (katajakasa) [inline src/game/gui/textinput.c:23]: could this be just "str text", no need to malloc ?

File: src/game/gui/textinput.c [hunk 2]
```diff
@@ -61,7 +61,7 @@ static void textinput_render(component *c) {
         mode = TEXT_DISABLED;
     }
 
-    text_render(&tb->tconf, mode, c->x, c->y, c->w, c->h, tb->buf);
+    text_render(&tb->tconf, mode, c->x, c->y, c->w, c->h, str_c(tb->text));
```
Comment (katajakasa) [inline src/game/gui/textinput.c:64]: text_render_str(), no need to call str_c().

File: src/game/gui/textinput.c (hunk 1 above)
Comment (mrannanj) [reply src/game/gui/textinput.c]: Changed to just str.

File: src/game/gui/textinput.c (hunk 2 above)
Comment (mrannanj) [reply src/game/gui/textinput.c]: Changed to text_render_str().

File: src/utils/str.h [hunk 3]
```diff
@@ -223,6 +223,11 @@ bool str_set_at(str *string, size_t pos, char value);
  */
 bool str_insert_at(str *string, size_t pos, char value);
 
+/**
+ * @brief Insert a C string starting at the given index.
+ */
+bool str_insert_c_at(str *dst, size_t pos, char *src);
```
Comment (katajakasa) [inline src/utils/str.h:229]: const char *src

Comment (mrannanj) [reply src/utils/str.h]: Added const.

**Comment by mrannanj** [comment]
I marked this as draft again in order to add more str tests.

**Pull request merged**